root/edit/details/EditPlace.js

```diff
@@ -0,0 +1,134 @@
+  const area = display.area;
+  const coordinates = display.coordinates;
+  const beginDate = display.begin_date;
+  const endDate = display.end_date;
+  const ended = display.ended;
+  const oldTypeName = display.type?.old?.name ?? '';
+  const newTypeName = display.type?.new?.name ?? '';
+  return (
+    <table className="details edit-place">
+      <tbody>
+        <tr>
+          <th>{addColonText(l('Place'))}</th>
+          <td><DescriptiveLink entity={place} /></td>
+        </tr>
+        {display.name ? (
+          <Diff
+            label={addColonText(l('Name'))}
+            newText={display.name.new}
+            oldText={display.name.old}
+            split="\s+"
+          />
+        ) : null}
+        {display.comment ? (
+          <Diff
+            label={addColonText(l('Disambiguation'))}
+            newText={display.comment ? display.comment.new : ''}
+            oldText={display.comment ? display.comment.old : ''}
+            split="\s+"
+          />
+        ) : null}
+        {display.type ? (
+          <FullChangeDiff
+            label={addColonText(l('Type'))}
+            newContent={newTypeName ? lp_attributes(newTypeName,
+                                                    'place_type') : ''}
+            oldContent={oldTypeName ? lp_attributes(oldTypeName,
+                                                    'place_type') : ''}
+          />
+        ) : null}
+        {address ? (
+          <Diff
+            label={addColonText(l('Address'))}
+            newText={address.new ? address.new : ''}
+            oldText={address.old ? address.old : ''}
+            split="\s+"
+          />
+        ) : null}
+        {area && area.new?.gid !== area.old?.gid ? (
+          <FullChangeDiff
+            label={addColonText(l('Area'))}
+            newContent={area.new
+              ? <DescriptiveLink entity={area.new} />
+              : ''}
+            oldContent={area.old
+              ? <DescriptiveLink entity={area.old} />
+              : ''}
+          />
+        ) : null}
+        {coordinates ? (
+          <Diff
+            label={addColonText(l('Coordinates'))}
+            newText={formatCoordinates(coordinates.new)}
+            oldText={formatCoordinates(coordinates.old)}
+          />
+        ) : null}
+        {beginDate ? (
+          <Diff
+            label={addColonText(l('Begin date'))}
+            newText={formatDate(beginDate.new)}
+            oldText={formatDate(beginDate.old)}
+            split="-"
+          />
+        ) : null}
+        {endDate ? (
+          <Diff
+            label={addColonText(l('End date'))}
+            newText={formatDate(endDate.new)}
+            oldText={formatDate(endDate.old)}
+            split="-"
+          />
+        ) : null}
+        {ended ? (
+          <FullChangeDiff
+            label={addColonText(l('Ended'))}
+            newContent={yesNo(ended.new)}
+            oldContent={yesNo(ended.old)}
+          />
```

Comment:
There's an issue here where "ended" diffs aren't currently displayed. Looks like this is because we don't convert those to booleans in `build_display_data`, so something like `{ new: 1, old: '0' }` will cause both sides to be true. (It worked in Perl 'cause `'0'` is falsey in Perl, but not so in JS.) So you'll need something like this in `build_display_data`:

```Perl
if (exists $self->data->{new}{ended}) {
    $data->{ended} = {
        new => boolean_to_json($self->data->{new}{ended}),
        old => boolean_to_json($self->data->{old}{ended}),
    };
}
```